job.go

```diff
@@ -207,3 +207,18 @@ func (j *Job) String() string {
 		"{project=%s params=%s group=%s id=%s}",
 		j.Project, j.Params, j.Group, j.ID[:7])
 }
+
+// State determines the job's current state by using it's path in the filesystem.
+func (j *Job) State(path, project, id string) (string, error) {
+	pPath := filepath.Join(path, project, "pending", id)
+	rPath := filepath.Join(path, project, "ready", id)
+	_, err := os.Stat(pPath)
```

Comment:
Instead for checking through the filesystem, which is racy if other methods of mistry manipulate the same paths, you can use the `JobQueue` struct which the server already contains.

Reply:
Good catch, but while I agree that this is racy, what would happen when we restart the server?The `JobQueue` would be empty then and the index page wouldn't display any jobs even if there are some. That's why I believe we should scan the filesystem.

Reply:
Let's revisit this at the next iteration of the web view. It's fine for now.